src/mnemonic/mod.rs

```diff
@@ -15,12 +19,26 @@ fn get_wordlist(language: Language) -> WordList {
     }
 }
 
-/// Converts a 12 word mnemonic to a entropy that can be used to
+/// Converts a 12 or 24 word mnemonic to entropy that can be used to
 /// generate a keypair
-pub fn mnemonic_to_entropy(words: Vec<String>) -> Result<[u8; 32]> {
-    if words.len() != 12 {
-        bail!("Invalid number of seed words");
-    }
+pub fn mnemonic_to_entropy(words: Vec<String>, seed_type: SeedType) -> Result<[u8; 32]> {
+    match seed_type {
+        SeedType::Bip39 => {
+            if words.len() != 12 && words.len() != 24 {
```

Comment:
huh, so bip39 allows for 12 word seed strings? I wasn't aware of that at all

Reply:
Heh. I thought the same thing. Its 12, 15, 18, 21, or 24. Any multiple of 3 from 12-24. AFAICT, 12 and 24 are by far the most common, and given the 256-bit entropy the wallet already wanted, easiest to implement.